shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizFeature.kt

```diff
@@ -46,6 +46,7 @@ interface StepQuizFeature {
             val attempt: Attempt,
             val submissionState: SubmissionState,
             val isProblemsLimitReached: Boolean,
+            val problemsLimitReachedModalText: String?,
```

Comment:
I suggest passing just a stepsLimitTotal (not a formatted string) from the dispatcher to do a viewMapping on the platform side, not in the dispatcher.
Ideally, viewMapping should be done in the viewStateMapper, but it's out of the scope of this task.

Reply:
@XanderZhu actually, we used this approach many times before (`ShowTopicCompletedModal`, `ShowStageCompletedModal`, `ShowProblemOfDaySolvedModal`) - I don't think that it's a problem to pass ready text, in this way we can avoid duplication texts on platforms. I even remember that on one of tech meetings we discussed that strings that required to be formatted with some data should be formatted in shared module to avoid code duplication. I leave that implementation as it is, we can discuss this one tech meeting - please, feel free to write me about it in slack 🙌